### ISSUE TITLE
[Tests] `test_save_chart_non_default_dir()` keeps failing 

### ISSUE THREAD

**Issue by nautics889** [gventuri/pandas-ai#661]
### Context
* PandasAI — latest (87dd966)
* Platform — Windows-based (Windows 11)
* Python — v3.11.4

### Description

I've noticed one of test case keeps failing when i run tests for `TestSmartDataframe` case. Exactly at this assert [statement](https://github.com/gventuri/pandas-ai/blob/87dd966e52c21e6359f2797f80826cd49c6bd40f/tests/test_smartdataframe.py#L469).

The diff:
_expected:_
```
'exports/charts/a0c00697-7468-4460-a36e-056d69fbd1ab.png'
```
_actual:_
```
'exports/charts\x070c00697-7468-4460-a36e-056d69fbd1ab.png'
```

At first i thought this one is similar to what was described in https://github.com/gventuri/pandas-ai/issues/347, but actually it seems quite different.

**Comment by gventuri** [comment]
@nautics889 great that you can reproduce it. It's not only a bug in the test. For some reason, windows is converting the slash to a backslash as we try to generate the path to it. I guess it's been there at least since 1.3.3.

**Comment by nautics889** [comment]
@gventuri same i thought, that the point is only in system's path-separator differencem, but these backslash looks more like byte sequence (e. g. `"\x00"`) as seems to me... Have to investigate this a bit, working on...

**Comment by gventuri** [comment]
Closing as duplicate of #663

Also, we had an issue in the CI that has been addressed by the PR #664 (fyi @nautics889, tests were silently failing).
